src/components/TransformWorkflow/FlowEditor/Components/OperationPanel/Forms/FlattenJsonOpForm.tsx

```diff
@@ -0,0 +1,222 @@
+  item: SxProps;
+} = {
+  container: {
+    border: '1px solid #F9F9F9',
+    color: '#5E5E5E',
+    alignItems: 'center',
+  },
+  headerItem: {
+    background: '#F9F9F9',
+    padding: '9px 16px 9px 16px',
+  },
+  item: {
+    border: '1px solid #F9F9F9',
+    padding: '9px 16px 9px 16px',
+  },
+};
+
+const FlattenJsonOpForm = ({
+  node,
+  operation,
+  sx,
+  continueOperationChain,
+  clearAndClosePanel,
+}: OperationFormProps) => {
+  const { data: session } = useSession();
+  const [srcColumns, setSrcColumns] = useState<ColumnData[]>([]);
+  const [jsonColumns, setJsonColumns] = useState<string[]>([]);
+  const globalContext = useContext(GlobalContext);
+  const nodeData: any =
+    node?.type === SRC_MODEL_NODE
+      ? (node?.data as DbtSourceModel)
+      : node?.type === OPERATION_NODE
+      ? (node?.data as OperationNodeData)
+      : {};
+
+  const { control, handleSubmit, reset } = useForm({
+    defaultValues: {
+      json_column: '',
+    },
+  });
+
+  const fetchAndSetSourceColumns = async () => {
+    if (node?.type === SRC_MODEL_NODE) {
+      try {
+        const columnData: ColumnData[] = await httpGet(
+          session,
+          `warehouse/table_columns/${nodeData.schema}/${nodeData.input_name}`
+        );
+        setSrcColumns(columnData);
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    if (node?.type === OPERATION_NODE) {
+      console.log(nodeData, 'node data');
+      setSrcColumns(
+        nodeData.output_cols.map((column: any) => ({ name: column }))
+      );
+    }
+  };
+
+  const fetchJsonColumns = async (selectedColumn: string) => {
+    try {
+      const response = await httpGet(
+        session,
+        `warehouse/dbt_project/json_columnspec/?source_schema=${nodeData.schema}&input_name=${nodeData.input_name}&json_column=${selectedColumn}`
+      );
+      console.log(response, 'json columns response');
+      setJsonColumns(response);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    fetchAndSetSourceColumns();
+  }, [session]);
+
+  const handleSave = async (formData: any) => {
+    try {
+      const inputUuid = node?.type === SRC_MODEL_NODE ? node?.data.id : '';
+      console.log('input_uuid:', inputUuid);
+      const sourceSchema = nodeData?.schema;
+
+      // Get the selected JSON column value from the form data
+      const selectedJsonColumn = formData.json_column;
+
+      // Fetch JSON columns based on the selected JSON column
+      await fetchJsonColumns(selectedJsonColumn);
+
+      // Make the API call
+      const operationNode: any = await httpPost(
+        session,
+        `transform/dbt_project/model/`,
+        {
+          op_type: operation.slug,
+          source_columns: srcColumns.map((column) => column.name),
+          other_inputs: [],
+          config: {
+            json_column: selectedJsonColumn,
+            source_schema: sourceSchema,
+            json_keys: jsonColumns,
```

Comment:
rename to `json_columns_to_copy` to avoid requiring a change in `dbt-automation`